[feat]: [CDS-91474]: Add Ignore status code for HTTP connections settings in the doc (#5763)
harness/developer-hub: docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/http-step.md

Capability checks are basic accessibility checks and do not follow multiple redirects. Hence, Harness returns from the first `302 Found` response during capability checks. 

:::info

> You can enable **Ignore status code for HTTP connections**, when enabled Harness only requires a valid response from the target HTTP server and does not verify the response code. This is particularly useful when the Harness delegate is configured with a proxy, as socket connection tests conducted by Harness from the delegate do not account for proxy details.

Reviewer: ```suggestion
You can enable **Ignore status code for HTTP connections**. When enabled, Harness only requires a valid response from the target HTTP server and does not verify the response code. This is particularly useful when the Harness Delegate is configured with a proxy because socket connection tests conducted by Harness from the delegate do not account for proxy details.
```

This setting is only relevant for HTTP steps and HTTP Helm repositories.

> You can enable the setting by following below steps:-

Reviewer: ```suggestion
You can enable the setting by following below steps:
```

> 1. Enable the setting by navigating to **Project Settings**. 

Reviewer 2: setting is available on ORG and ACCOUNT level as well

Author: Okay I will add that as well

2. Select **Default Settings** under **General**.
3. Under **Continuous Deployment**, you can set **Ignore status code for HTTP connections** as **true**.
![](./static/Ignore%20status%20code%20for%20HTTP%20connections.png)

> This setting can be enbaled using feature flag ``CDS_USE_HTTP_CHECK_IGNORE_RESPONSE_INSTEAD_OF_SOCKET_NG``.

Reviewer: ```suggestion
This setting is behind the feature flag, ``CDS_USE_HTTP_CHECK_IGNORE_RESPONSE_INSTEAD_OF_SOCKET_NG``. To enable a feature flag in your Harness account, contact [Harness Support](mailto:support@harness.io).
```

> This setting is behind the feature flag, ``CDS_USE_HTTP_CHECK_IGNORE_RESPONSE_INSTEAD_OF_SOCKET_NG``. To enable this feature flag in your Harness account, contact [Harness Support](mailto:support@harness.io).

Reviewer 2: setting is independent of the feature flag, it will always be available. To use this feature currently, we need EITHER of the FF or setting to be enabled. 
We should mention that we are going to remove the FF and this feature should only be controlled by the setting going forward.

Author: Sure I will add that as a note as well
